products/views.py

```diff
@@ -1,3 +1,54 @@
+import requests
+
 from django.shortcuts import render
+from django.http      import JsonResponse
+
+from django.views    import View
+from products.models import (
+    Product,
+    SubCategory,
+    Category
+)
+from users.models    import (
+    User,
+    Creator
+)
+
+class GetList(View):
+    def get(self, request):
+
+        filter_type     = request.GET.get("type", None)
+        sub_category_id = request.GET.get("sub_category", None)
+
+        num = request.GET.get("number", 20)
+
+        filter_dict = {}
+        order_dict = {'top10' : '-likes', 'new' : '-created_at'}
+
+        order_condition = order_dict.get(filter_type, 'id')
+
+        if sub_category_id:
+            filter_dict["sub_category_id"] = sub_category_id
+
+        products = Product.objects.select_related(
+            'sub_category',
+            'creator__user'
+        )
+
+        products = products.filter(**filter_dict).order_by(order_condition)[:int(num)][::1]
```

Comment:
코드 정렬해주시고, filter_dict도 미리 선언하지 않으셔도 될 것 같습니다.
```suggestion
        filter_type     = request.GET.get("type", None)
        sub_category_id = request.GET.get("sub_category", None)
        num = request.GET.get("number", 20)

        filter_dict = {}
        order_dict = {'top10' : '-likes', 'new' : '-created_at'}
        order_condition = order_dict.get(filter_type, 'id')

        if sub_category_id:
            filter_dict["sub_category_id"] = sub_category_id

        products = Product.objects.select_related(
            'sub_category',
            'creator__user'
        )

        products = products.filter(**filter_dict).order_by(order_condition)[:int(num)][::1]
```